### PR TITLE
add ID to DB/orders & serch order by ID

### PR DESCRIPTION
**Pull request by sergklukvin**
add ID to DB/orders & serch order by ID

### CODE REVIEW

**Review by YuriiDubnytskyi — APPROVED** [review]

File: modules/order/order.service.js [hunk 1]
```diff
@@ -133,21 +131,23 @@ class OrdersService {
...
-      { ...order, lastUpdatedDate: Date.now() },
+      { ...orderUpdate, lastUpdatedDate: Date.now() },
       { new: true }
     ).exec();
   }
 
-  async addOrder(data) {
+  async addOrder(data, { user }) {
     const { items } = data;
+    const { _id } = user;
```
Comment (AndriyLymych) [inline modules/order/order.service.js:149]: when user is unauthorized, you cant get his ID

File: modules/order/order.resolver.js [hunk 2]
```diff
@@ -17,8 +17,14 @@ const ordersQuery = {
     };
   },
   getAllOrders: async (parent, args) => await ordersService.getAllOrders(args),
-  getUserOrders: async (parent, args, context) =>
-    await ordersService.getUserOrders(context.user),
+  getUserOrders: async (parent, args, context) => {
```
Comment (AndriyLymych) [inline modules/order/order.resolver.js:20]: please, use destructuring

File: modules/order/order.resolver.js [hunk 3]
```diff
@@ -33,9 +39,9 @@ const ordersQuery = {
 };
 
 const ordersMutation = {
-  addOrder: async (parent, args) => {
+  addOrder: async (_, { order }, context) => {
```
Comment (AndriyLymych) [inline modules/order/order.resolver.js:42]: same

File: modules/order/order.service.js [hunk 4]
```diff
@@ -108,9 +105,10 @@ class OrdersService {
   }
 
   async getOrderById(id) {
-    if (!ObjectId.isValid(id)) throw new Error(ORDER_NOT_VALID);
+    if (!ObjectId.isValid(id)) throw new Error(ORDER_NOT_FOUND);
```
Comment (AndriyLymych) [inline modules/order/order.service.js:108]: use RuleError

File: modules/order/order.service.js [hunk 5]
```diff
@@ -108,9 +105,10 @@ class OrdersService {
...
 
     const foundOrder = await Order.findById(id).exec();
+
     if (!foundOrder) throw new Error(ORDER_NOT_FOUND);
```
Comment (AndriyLymych) [inline modules/order/order.service.js:112]: same

File: modules/order/order.service.js [hunk 6]
```diff
@@ -178,10 +178,12 @@ class OrdersService {
     return foundOrder;
   }
 
-  async getUserOrders(user) {
-    const { orders } = user;
+  async getUserOrders({ id }) {
+    const userOrders = await Order.find({ 'user.id': id }).exec();
+
+    if (!userOrders) throw new Error(ORDER_NOT_FOUND);
```
Comment (AndriyLymych) [inline modules/order/order.service.js:184]: same

**Comment by AndriyLymych** [comment]
Also, add permissions for getting order by user , please

File: modules/order/order.resolver.js (hunk 2 above)
Comment (sergklukvin) [reply modules/order/order.resolver.js:20]: fixed

File: modules/order/order.resolver.js (hunk 3 above)
Comment (sergklukvin) [reply modules/order/order.resolver.js:42]: fixed

File: modules/order/order.resolver.js [hunk 7]
```diff
@@ -33,9 +39,9 @@ const ordersQuery = {
...
     try {
-      return await ordersService.addOrder(args.order);
+      return await ordersService.addOrder(order, context);
     } catch (e) {
       return {
```
Comment (grebiniuk) [inline modules/order/order.resolver.js:46]: It must be a RuleError(...)

**Review by grebiniuk — CHANGES_REQUESTED** [review]

File: modules/order/order.service.js (hunk 4 above)
Comment (sergklukvin) [reply modules/order/order.service.js:108]: fixed

File: modules/order/order.service.js (hunk 5 above)
Comment (sergklukvin) [reply modules/order/order.service.js:112]: fixed

File: modules/order/order.service.js (hunk 6 above)
Comment (sergklukvin) [reply modules/order/order.service.js:184]: fixed

File: modules/order/order.resolver.js (hunk 7 above)
Comment (sergklukvin) [reply modules/order/order.resolver.js:46]: fixed

File: modules/order/order.service.js [hunk 8]
```diff
@@ -144,10 +142,15 @@ class OrdersService {
     ).exec();
   }
 
-  async addOrder(data, { user }) {
+  async addOrder(data, user) {
     const { items } = data;
-    const { _id } = user;
-    data.user = { ...data.user, id: _id };
+
+    if (typeof user !== 'undefined') {
```
Comment (grebiniuk) [inline modules/order/order.service.js:148]: it will not work:
image

Comment (sergklukvin) [reply modules/order/order.service.js]: fixed

**Review by YuriiDubnytskyi — APPROVED** [review]

**Review by YuriiDubnytskyi — APPROVED** [review]

**Review by yuliiakucher — APPROVED** [review]

**Review by grebiniuk — APPROVED** [review]

**Pull request merged**